General | Clean and remove /DietPi/dietpi/conf

MichaIng opened the pull request:
**Status**: Ready

**Commit list/description**:
+ DietPi-PREP | Moved initial `/etc/network/interfaces` into PREP, since it is not used by any other script
+ General | Move default mpd.conf and cava.conf from RAMdisk to disk and remove obsolete `dietpi/conf` on update
+ DietPi-Patch | Minor coding: Apply bash argument duplication for removal/chmod from RAMdisk and disk location at once
+ DietPi-Config | Minor coding

MichaIng committed:
v6.20

+ DietPi-PREP | Moved initial "/etc/network/interfaces" into PREP. This will be re-created by DietPi-Config

MichaIng committed:
v6.20

+ DietPi-PREP | Moved initial /etc/network/interfaces into PREP

MichaIng commented:
@Fourdee 
What is the reason for `cava.conf` and `mpd.conf` remaining in RAMdisk? I can't find them being touched, besides within `dietpi-software` which requires network already.
- [x] If required without network access, we could move them to: `/var/lib/dietpi/dietpi-software/conf`?

`/etc/network/interfaces` is actually mostly created at three places now:
- DietPi-PREP
- DietPi-PreBoot
- DietPi-Config

I think we should add this to `DietPi-Set_Hardware` to apply settings from `dietpi.txt` to `/etc/network/interfaces`, also respecting AP choice, and setting defaults for non-existent values.
Then call this from PreBoot and Config and we don't need to pre-create anything in PREP, but can also call it as well from there.

Same could be done with some other configs, e.g. `crontab`.

MichaIng committed:
v6.20

+ General | Move default cava.conf from RAMdisk to disk

MichaIng committed:
v6.20

+ General | Move default mpd.conf from RAMdisk to disk

MichaIng committed:
v6.20

+ DietPi-Software | MPD/Cava: Copy default config from new disk location

MichaIng committed:
v6.20

+ DietPi-Patch | Remove obsolete dietpi/conf dir from RAMdisk and disk
+ DietPi-Patch | Minor coding: Apply bash argument duplication for removal/chmod from RAMdisk and disk location at once

MichaIng commented:
Quite nice to handle RAMdisk + disk location with one command:
```
2019-01-05 16:03:26 root@micha:/tmp# echo /tmp/{Atest1,Btest2}/test
/tmp/Atest1/test /tmp/Btest2/test
```
- Single curly braces can be generally used to duplicate arguments according to braces content. Quite handy, didn't know this one 😄.

`rm`, `chmod` and `chown` all have `-f` option to suppress file-missing errors.

MichaIng committed:
v6.20

+ DietPi-Config | Minor coding

MichaIng committed:
v6.20

+ CHANGELOG | Moved the remaining conf/* files located on DietPi-RAMdisk to /var/lib/dietpi or into DietPi-PREP

MichaIng commented on dietpi/dietpi-config:
```diff
@@ -367,13 +361,13 @@
 
-					G_WHIP_DEFAULT_ITEM="$i"
-					if G_WHIP_INPUTBOX "Please enter a value (pixel count) for ${overscan_options[$i]}.\n - EG: 16"; then
+					G_WHIP_DEFAULT_ITEM=${!i}
```
`${!i}` expends `$i` and then the value of variable `$i`:

EG:
- `$i == overscan_left`
- `$overscan_left == 15`
- => `${!i} == 15`

Fourdee commented:
~Unsure about cava, however, following is used by Allo GUI to reset configs on changes:~
~I'll need to update the GUI code.~

@MichaIng 

Just checked Allo GUI code, no matches for ```/DietPi/dietpi/conf```, good to go.

Fourdee commented:
- ```cava.conf```

Used in ```dietpi-justboom``` to reset config
```
		# - Reset CAVA config
		cp /DietPi/dietpi/conf/cava.conf $HOME/.config/cava/config

```

Commit:
v6.20

+ Update FP for cava.conf: https://github.com/Fourdee/DietPi/pull/2393

Commit:
v6.20

+ Minor, FP remove reference.

Commit:
v6.20

+ rm last reference.

Fourdee commented:
🈯️ We can move ```/var/lib/dietpi/dietpi-software/conf/mpd.conf ``` to github as download item.

Commit:
v6.20

+ move /var/lib/dietpi/dietpi-software/conf/mpd.conf to github as download item.

Fourdee commented:
@MichaIng 

Great work, looks good, will merge 👍

MichaIng commented:
@Fourdee 
> Used in dietpi-justboom to reset config

Damn, doing repo (code) searches on GitHub for file paths does not work very well. Slash is not taken literally, even with espacing.
So good find.

About `/etc/network/interfaces`:
- Jep so currently I left/added default to PREP code, so default file will be in place on 1st boot.
- Moving it's (re)creation into separate DietPi-Set_Hardware (or dedicated) script is another midterm task, to have this consistent across scripts. I anyway had the idea to only add those contents that are required. So skip adding WiFi related entries, if only Ethernet is found/enabled and the other way round. I am just not yet sure where/how to safe or hand over WiFi and Ethernet adapter specific values, since `dietpi.txt` only allows IP/Mask/DNS for both together. Currently, if not WiFi hotspot enabled, we anyway only allow one interface enabled, but to allow values being saved, even if the interface is disabled, would be still good. Currently this is done within the interface file itself, which does not require any additional config/db file 🤔.
- However different topic.
